components/offer/MakingOfferSingleItem.vue

```diff
@@ -73,10 +72,12 @@ const { chainSymbol, decimals } = useChain()
 
 const item = computed(() => offerStore.items[0])
 
-const itemPrice = computed(() => formatWithBlank(Number(item.value.price)))
+const itemPrice = computed(() => formatWithBlank(item.value.price))
 
-const formatWithBlank = (value: number) => {
-  return value ? formatBalance(value, decimals.value, chainSymbol.value) : '--'
+const formatWithBlank = (value: string | undefined) => {
+  return value
+    ? useAmount(computed(() => value), decimals, chainSymbol).formatted
+    : '--'
```

Comment:
DRY lets avoid this code duplication

I was suggesting something like 

````

export default function (
  tokenAmount: ComputedRef<number | string | undefined>,
  tokenDecimals: ComputedRef<number>,
  chainSymbol: ComputedRef<string>,
  { roundBy, withBlank = false }: { roundBy?: ComputedRef<Prefix | number>, withBlank?: boolean } = {}
) {
  const { getCurrentTokenValue } = useFiatStore()

  const amountFormatted = computed(() => {
    const amount = tokenAmount.value
      ? formatAmountWithRound(tokenAmount.value, tokenDecimals.value, roundBy?.value)
      : 0

    if (amount && withBlank) {
      return '--'
    }

    return `${amount} ${chainSymbol.value}`
  })

  const amountUsd = computed(() => {
    const value = calculateUsdFromToken(
      Number(tokenAmount.value) * Math.pow(10, -tokenDecimals.value),
      Number(getCurrentTokenValue(chainSymbol.value)),
    )
    return typeof value === 'number' && value >= 0 ? `$${value}` : ''
  })

  return {
    formatted: amountFormatted,
    usd: amountUsd,
  }
}

````

so you can do 

```suggestion
  return useAmount(computed(() => value), decimals, chainSymbol, { withBlank: true } ).formatted.value
```

Reply:
I'm sure you meant `if (!amount && withBlank) {
      return '--'
    }` ?
    Doesn't it make more sense to just check for amount falsy and additional blank option and just return '--' straight up and then proceed

Reply:
> I'm sure you meant `if (!amount && withBlank) { return '--' }` ?

yes mb

> Doesn't it make more sense to just check for amount falsy and additional blank option and just return '--' straight up and then proceed

you don't want to have '--' in all instaces, sometimes you do want to show that value is '0 DOT' for example

Reply:
oh yeah of course that's what the blank option is for. Might have worded that wrong.